### ISSUE TITLE
[no-duplicate-landmark-elements] false positive with content within {{#if}}/{{else}} – apply similar fix as with no-duplicate-id?

### ISSUE THREAD

**Issue by johanrd** [ember-template-lint/ember-template-lint#3153]
The no-duplicate-landmark-elements rule in ember-template-lint currently flags conditionally rendered landmark elements as errors, even when only one instance of the element will be present in the DOM at any given time. This is a false positive, similar to the issue that was previously addressed for the no-duplicate-ids here: https://github.com/ember-template-lint/ember-template-lint/issues/1511

## Reproduction Case

Consider the following Glimmer component template:

```hbs
{{#if this.isDesktop}}
  <header aria-label="Desktop Header"></header>
{{else}}
  <header aria-label="Mobile Header" ></header>
{{/if}}
```

In this example, there are two <header> elements, but they are mutually exclusive. Only one will be rendered based on the (media 'isDesktop') condition. However, ember-template-lint still flags this as an error. 

## Proposed solution
This general concept seems to have been fixed in no-duplicate-id, see: https://github.com/ember-template-lint/ember-template-lint/pull/1606

**Comment by lifeart** [comment]
Closing as issue should be fixed in #3152

**Comment by johanrd** [comment]
@lifeart the above reproduction is still reproducible after issue #3152. The PR #3176 seems to only have addressed the landmark context of dialogs/popovers, not mutually exclusive `{{#if}}/{{else}}` blocks, right?

```
<template>
  {{#if @isDesktop}}
    <header>Desktop</header>
  {{else}}
    <header>Mobile</header>
  {{/if}}
</template>
```

**Comment by lifeart** [comment]
Hey @johanrd! Additional fixes landed in https://github.com/ember-template-lint/ember-template-lint/pull/3201
Your snippet works well now

<img width="1310" alt="Image" src="https://github.com/user-attachments/assets/fc969632-3440-4231-982d-5484471435b5" />

**Comment by johanrd** [comment]
@lifeart ah, thanks, you are correct!
